src/components/Creation/Custom/index.tsx

```diff
@@ -1,3 +1,140 @@
-export default function Custom() {
-  return <div></div>;
+type CustomTypes = 'background' | 'character' | 'sticker';
+
+type CustomProps = {
+  setSelectedBackground: (item: number | null) => void;
+  setSelectedCharacter: (item: number | null) => void;
+  setSelectedSticker: (item: number | null) => void;
+  selectedItem: CustomTypes;
+  setSelectedItem: (item: CustomTypes) => void;
+};
+
+type CustomItem = {
+  name: string;
+  value: CustomTypes;
+};
+
+const CUSTOM_ITEMS: CustomItem[] = [
+  { name: '배경', value: 'background' },
+  { name: '캐릭터', value: 'character' },
+  { name: '스티커', value: 'sticker' },
+];
+
+const MOCK_IMAGES = [
+  {
+    id: 0,
+    value: 'url1',
+  },
+  {
+    id: 1,
+    value: 'url2',
+  },
+  {
+    id: 2,
+    value: 'url3',
+  },
+  {
+    id: 3,
+    value: 'url1',
+  },
+  {
+    id: 4,
+    value: 'url2',
+  },
+  {
+    id: 5,
+    value: 'url3',
+  },
+];
+
+export default function Custom(props: CustomProps) {
+  const {
+    setSelectedBackground,
+    setSelectedCharacter,
+    setSelectedSticker,
+    selectedItem,
+    setSelectedItem,
+  } = props;
+
+  const handleCustomTypeClick = (value: CustomTypes) => {
+    setSelectedItem(value);
+  };
+
+  const handleItemClick = (id: number) => {
+    if (selectedItem === 'background') {
+      setSelectedBackground(id);
+      setSelectedCharacter(null);
+      setSelectedSticker(null);
+    }
+    if (selectedItem === 'character') {
+      setSelectedCharacter(id);
+      setSelectedBackground(null);
+      setSelectedSticker(null);
+    }
+    if (selectedItem === 'sticker') {
+      setSelectedSticker(id);
+      setSelectedCharacter(null);
+      setSelectedBackground(null);
+    }
+  };
+
+  return (
+    <div className="w-96 h-96 bg-blossom-lightBlue flex-col space-y-2">
+      <div className="grid grid-cols-3 gap-4 pl-5 pr-3 mx-1">
+        {CUSTOM_ITEMS.map(custom => {
+          if (selectedItem === custom.value) {
+            return (
+              <button
+                key={custom.value}
+                className="w-24 h-9 text-sm rounded bg-blossom-green border-blossom-white border-2"
+                onClick={() => handleCustomTypeClick(custom.value)}
+              >
+                {custom.name}
+              </button>
+            );
+          }
+          if (selectedItem !== custom.value) {
+            return (
+              <button
+                key={custom.value}
+                className="w-24 h-9 text-sm rounded bg-blossom-yellow border-blossom-white border-2"
+                onClick={() => handleCustomTypeClick(custom.value)}
+              >
+                {custom.name}
+              </button>
+            );
+          }
+        })}
```

Comment:
return 값이 배경색말고 전부 똑같네요. 그러면 그냥 간단하게 처리하는게 낫지 않을까요?
```suggestion
        {CUSTOM_ITEMS.map(custom => {
            return (
              <button
                key={custom.value}
                className={`w-24 h-9 text-sm rounded border-blossom-white border-2 
                ${selectedItem === custom.value ? 'bg-blossom-green' : 'bg-blossom-yellow'}`}
                onClick={() => handleCustomTypeClick(custom.value)}
              >
                {custom.name}
              </button>
            );
          }
        })}
```

Reply:
제가 tailwind에서 동적으로 색을 바꾸는 방법을 못찾아서 해당 방법으로 처리했는데 혹시 바꿀 수 있는 방법이 있을까요?

Reply:
제가 올린대로 하시면 됩니당. 클래스명이 온전해야돼요

Reply:
아하 넵!! 감사합니다